src/main.rs

```diff
@@ -113,36 +114,40 @@ fn run_once(
     let mut messages: Vec<Message> = paths
         .par_iter()
         .map(|entry| {
-            lint_path(entry.path(), settings, &cache.into(), &autofix.into()).unwrap_or_else(|e| {
-                if settings.select.contains(&CheckCode::E999) {
-                    vec![Message {
-                        kind: CheckKind::SyntaxError(e.to_string()),
-                        fixed: false,
-                        location: Default::default(),
-                        filename: entry.path().to_string_lossy().to_string(),
-                    }]
+            match entry {
+                Ok(entry) => {
+                    let path = entry.path();
+                    lint_path(path, settings, &cache.into(), &autofix.into())
+                        .map_err(|e| (Some(path.to_owned()), e.to_string()))
+                }
+                Err(e) => Err((
+                    e.path().map(Path::to_owned),
+                    e.io_error()
+                        .map_or_else(|| e.to_string(), io::Error::to_string),
+                )),
+            }
+            .unwrap_or_else(|(path, message)| {
+                if let Some(path) = path {
+                    if settings.select.contains(&CheckCode::E902) {
+                        vec![Message {
+                            kind: CheckKind::IOError(message),
+                            fixed: false,
+                            location: Default::default(),
+                            filename: path.to_string_lossy().to_string(),
+                        }]
+                    } else {
+                        error!("Failed to check {}: {message}", path.to_string_lossy());
+                        vec![]
+                    }
                 } else {
-                    error!("Failed to check {}: {e:?}", entry.path().to_string_lossy());
+                    error!("{message}");
                     vec![]
                 }
             })
         })
         .flatten()
         .collect();
 
-    if settings.select.contains(&CheckCode::E902) {
-        for file in files {
-            if !file.exists() {
-                messages.push(Message {
-                    kind: CheckKind::IOError(file.to_string_lossy().to_string()),
-                    fixed: false,
-                    location: Default::default(),
-                    filename: file.to_string_lossy().to_string(),
-                })
-            }
-        }
-    }
```

Comment:
I think this was present specifically to handle the user explicitly providing a path to a file that doesn't exist. Are we still raising an error, similar to Flake8?

```
∴ flake8 foo.py
foo.py:0:1: E902 FileNotFoundError: [Errno 2] No such file or directory: 'foo.py'
```

Reply:
Yep.

```
$ ruff foo.py
Found 1 error(s).
foo.py:0:0: E902 No such file or directory (os error 2)
```